src/ChainRules.jl

```diff
@@ -13,22 +22,32 @@ if VERSION < v"1.3.0-DEV.142"
     import LinearAlgebra: dot
 end
 
-import NaNMath, SpecialFunctions
-
-export AbstractRule, Rule, frule, rrule
-
-include("differentials.jl")
-include("rules.jl")
-include("rules/base.jl")
-include("rules/array.jl")
-include("rules/broadcast.jl")
-include("rules/mapreduce.jl")
-include("rules/linalg/utils.jl")
-include("rules/linalg/blas.jl")
-include("rules/linalg/dense.jl")
-include("rules/linalg/structured.jl")
-include("rules/linalg/factorization.jl")
-include("rules/nanmath.jl")
-include("rules/specialfunctions.jl")
+include("helper_functions.jl")
+
+include("rulesets/Base/base.jl")
+include("rulesets/Base/array.jl")
+include("rulesets/Base/broadcast.jl")
+include("rulesets/Base/mapreduce.jl")
+
+include("rulesets/LinearAlgebra/utils.jl")
+include("rulesets/LinearAlgebra/blas.jl")
+include("rulesets/LinearAlgebra/dense.jl")
+include("rulesets/LinearAlgebra/structured.jl")
+include("rulesets/LinearAlgebra/factorization.jl")
+
+# Note: The following is only required because package authors do not use
```

Comment:
Grammar. Not entirely clear on what you mean here.

Reply:
updated